packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx

```diff
@@ -912,14 +912,18 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
       case Downshift.stateChangeTypes.keyDownEnter:
       case Downshift.stateChangeTypes.clickItem:
         const shouldAddHighlightedIndex = !multiple && items && items.length > 0;
+        const isSameItemSelected = changes.selectedItem === undefined;
```

Comment:
```
const isSameItemSelected = changes.selectedItem === undefined;
const newValue = isSameItemSelected ? value[0] : changes.selectedItem;

newState.open = false;
newState.searchQuery = this.getSelectedItemAsString(newValue);
if (!isSameItemSelected) {
    newState.value = multiple ? [...value, changes.selectedItem] : [changes.selectedItem];
    newState.highlightedIndex = shouldAddHighlightedIndex ? items.indexOf(newValue) : null;

    if (getA11ySelectionMessage && getA11ySelectionMessage.onAdd) {
      this.setA11ySelectionMessage(getA11ySelectionMessage.onAdd(changes.selectedItem));        
    }
}
```
_nit_
Would having `newState.value`, `newState.highlightedIndex` and `this.setA11ySelectionMessage` inside `! isSameItemSelected` condition work the same?

Reply:
cool! it might work, let me try!